src/fontique/backend/android.rs

```diff
@@ -87,11 +94,59 @@ impl SystemFonts {
                                         //       smarter, or something dumb that meets expecta­
                                         //       tions on Android.
                                     }
-                                } else if let Some(_langs) = child
+                                } else if let Some(langs) = child
                                     .attribute("lang")
                                     .map(|s| s.split(',').collect::<Vec<&str>>())
                                 {
-                                    // TODO: implement language fallback "family" elements
+                                    let (_has_for, hasnt_for): (Vec<Node>, Vec<Node>) = child
+                                        .children()
+                                        .partition(|c| c.attribute("fallbackFor").is_some());
+                                    {
+                                        // general fallback families
+                                        let (ps_named, _ps_unnamed): (Vec<Node>, Vec<Node>) =
+                                            hasnt_for.iter().partition(|c| {
+                                                c.attribute("postScriptName").is_some()
+                                            });
+
+                                        if let Some(family) = ps_named.iter().find_map(|x| {
+                                            postscript_names
+                                                .get(x.attribute("postScriptName").unwrap())
+                                        }) {
+                                            for lang in langs {
+                                                if let Some(scr) = lang.strip_prefix("und-") {
```

Comment:
Can we have a comment expanding `und` here please? Trying to search for it only brings up the README for https://github.com/behdad/fontconfig, which is not at all enlightening

Reply:
`und` is just a BCP-47 language subtag for “undefined”. It’s used when fallback only depends on script.

Reply:
I second @DJMcNab's suggestion that this is the sort of thing that belongs in the code as a comment. I think we should be aiming for the entire codebase to be explained in such detail, along with a higher-level architectural explanation in a (markdown?) doc (such that it's approachable for someone who doesn't know much if anything about text, but is willing to learn). Obviously it's going to take a while to backfill that for existing code, but we could probably do it more easily for new code.

Perhaps others consider this unnecessary or onerous to the point that it will prevent people from contributing? But if I was reviewing code at work, that's the standard I would expect. And I also feel like it's what I will need to be able to meaningfully contribute to this project (given enough time, I might be able to do a "deep reading" of the code and fill in some this myself, but even that can be tricky around some of the stuff like font fallback and bidi resolution which involve tricky domain-specific knowledge - it's easy to end up understanding what the code is doing, but not *why*, which can make it difficult to work out how to safely modify the code without breaking things).